src/abstract-interpretation/domain.ts

```diff
@@ -90,23 +90,41 @@ function compareIntervalsByTheirMaximum(interval1: Interval, interval2: Interval
 	return compareIntervals(CompareType.Max, interval1.max, interval2.max)
 }
 
-export function doIntervalsOverlap(interval1: Interval, interval2: Interval): boolean {
+export const enum OverlapKind {
+	Overlap = 0,
+	Touch = 1,
+}
+
+export function doIntervalsOverlap(interval1: Interval, interval2: Interval, kind: OverlapKind = OverlapKind.Overlap): boolean {
```

Comment:
We definitely need tests for this.

Reply:
and fundamentally i think the function would be simpler if handled as an intersection of interval, checking if it is empty/scalar.

Reply:
> We definitely need tests for this.

We already have some (six), but they don't yet cover all edge cases.

> and fundamentally i think the function would be simpler if handled as an intersection of interval, checking if it is empty/scalar.

This would be definitly more elegant and reusable. I'll write it down somewhere and change it when I get to it. For now, I prioritize making progress; refactorings like this can come later.

Reply:
This is now tracked in #588